onnx/defs/nn/defs.cc

```diff
@@ -2487,4 +2487,195 @@ ONNX_OPERATOR_SET_SCHEMA(
         )ONNX"
         ));
 
+void col2imShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+  // validating 'pads' attribute
+  std::vector<int64_t> pads;
+  if (getRepeatedAttribute(ctx, "pads", pads)) {
+    if (pads.size() != 2 * 2) {
+      fail_shape_inference("Attribute pads has incorrect size");
+    }
+  }
+
+  std::vector<int64_t> dilations;
+  if (getRepeatedAttribute(ctx, "dilations", dilations)) {
+    if (dilations.size() != 2) {
+      return;
+    }
+  }
+
+  std::vector<int64_t> strides;
+  if (getRepeatedAttribute(ctx, "strides", strides)) {
+    if (strides.size() != 2) {
+      return;
+    }
+  }
+
+  // validating 'block_shape' input
+  std::vector<int64_t> block_shape;
+  const TensorProto* temp_block_shape = ctx.getInputData(2);
+  if (temp_block_shape) {
+    block_shape = ParseData<int64_t>(temp_block_shape);
+    if (block_shape.size() != 2) {
+      fail_shape_inference("block_shape tensor has wrong rank. It must have rank 2.");
+    }
+  } else {
+    if (ctx.getInputType(2)->tensor_type().shape().dim_size() != 1) {
+      fail_shape_inference("block_shape tensor has wrong rank. It must have rank 2.");
```

Comment:
Do you mean `rank 1`?

Reply:
It is correct as it is.
`dim_size()` vs `dim_value()` has caused some confusion in previous reviews, but `ctx.getInputType(2)->tensor_type().shape().dim_size()` should return 1 *dimension* and `ctx.getInputType(2)->tensor_type().shape().dim(0).dim_value()` should return the *value held by first dim*, which should be 2

Reply:
Sorry, the if-condition is checking for 1 (which is correct), but the error message says `rank 2` (which is incorrect, IIUC).